src/lib/array.jl

```diff
@@ -21,48 +21,10 @@ end
 @adjoint (::Type{T})(sz) where {T<:Zeros} = T(sz), Δ->(nothing,)
 @adjoint (::Type{T})(sz) where {T<:Ones} = T(sz), Δ->(nothing,)
 
-@adjoint getindex(x::AbstractArray, inds...) = x[inds...], ∇getindex(x, inds)
-
-@adjoint view(x::AbstractArray, inds...) = view(x, inds...), ∇getindex(x, inds)
-
-∇getindex(x::AbstractArray{T,N}, inds) where {T,N} = dy -> begin
-  if inds isa NTuple{N,Int} && T <: Number
-    dx = OneElement(dy, inds, axes(x))
-  elseif inds isa NTuple{<:Any, Integer}
-    dx = _zero(x, typeof(dy))
-    dx[inds...] = dy
-  else
-    dx = _zero(x, eltype(dy))
-    dxv = view(dx, inds...)
-    dxv .= accum.(dxv, _droplike(dy, dxv))
-  end
-  return (_project(x, dx), map(_->nothing, inds)...)
-end
-
-"""
-    OneElement(val, ind, axes) <: AbstractArray
-
-Extremely simple `struct` used for the gradient of scalar `getindex`.
-"""
-struct OneElement{T,N,I,A} <: AbstractArray{T,N}
-  val::T
-  ind::I
-  axes::A
-  OneElement(val::T, ind::I, axes::A) where {T<:Number, I<:NTuple{N,Int}, A<:NTuple{N,AbstractUnitRange}} where {N} = new{T,N,I,A}(val, ind, axes)
-end
-Base.size(A::OneElement) = map(length, A.axes)
-Base.axes(A::OneElement) = A.axes
-Base.getindex(A::OneElement{T,N}, i::Vararg{Int,N}) where {T,N} = ifelse(i==A.ind, A.val, zero(T))
```

Comment:
Deleting this struct makes Flux fail. 

It could just be left in deprecations. Or it could be hooked up more narrowly to CR's rule, something like this shouldn't upset 2nd derivative definitions:

```
ChainRules.∇getindex(x::Array{<:Number,N}, dy, inds::Vararg{Integer,N}) where N =
  OneElement(dy, inds, axes(x))
```

But that's piracy...

Reply:
I didn't bother reverting the deletion into a deprecation because https://github.com/FluxML/Zygote.jl/pull/1328#issuecomment-1362394766 was still unresolved, but that seems like the way to go.

Reply:
Sure. To be clear I think returning one `nothing` is an upgrade even though it makes those tests fail.

If someone approves https://github.com/JuliaDiff/ChainRules.jl/pull/687 then perhaps CI can be made happy here.